text/0001-script-based-tokenizer.md

```diff
@@ -49,11 +49,81 @@ We should implement a new tokenizer which detect script and tokenize based on it
+
 ```rust
 let tokenizer = Tokenizer::new("The quick brown fox jumps over the lazy dog");
-assert_eq!(tokenizer.next().txt(), "The")
-assert_eq!(tokenizer.next().txt(), " ")
-assert_eq!(tokenizer.next().txt(), "quick")
+let token_group = tokenizer.next();
+assert_eq!(token_group.tokens().map(|t| t.text()).collect(), ["The"]);
+assert_eq!(token_group.normalized().map(|t| t.text()).collect(), ["the"]);
+let token_group = tokenizer.next();
+assert_eq!(token_group.tokens().map(|t| t.text()).collect(), [" "])
+let token_group = tokenizer.next();
+assert_eq!(token_group.tokens().map(|t| t.text()).collect(), ["quick"])
+```
+
+#### highlight
+
+```rust
+fn highlight_record(record: &mut IndexMap<String, String>, words: &HashSet<String>) {
+  for (_key, value) in record.iter_mut() {
+    let old_value = mem::take(value);
+    let tokenizer = Tokenizer::new(&the_string);
+    for token_group in tokenizer {
+      // get the longest normalized token
+      let token = token_goup.normalized().max_by(|a, b| a.token_len().cmp(b.token_len()));
+      if token.kind() == TokenKind::Word {
+        let normalized_token = token.text();
+        let to_highlight = words.contains(&normalized_token);
+        if to_highlight { value.push_str("<mark>") }
+        value.push_str(token.original.text());
+        if to_highlight { value.push_str("</mark>") }
+      } else {
+        value.push_str(token);
+      }
+    }
+  }
+}
+```
+
+```rust
+// original
+fn highlight_record(record: &mut IndexMap<String, String>, words: &HashSet<String>) {
+    for (_key, value) in record.iter_mut() {
+        let old_value = mem::take(value);
+        for (token_type, token) in simple_tokenizer(&old_value) {
+            if token_type == TokenType::Word {
+                let lowercase_token = token.to_lowercase();
+                let to_highlight = words.contains(&lowercase_token);
+                if to_highlight { value.push_str("<mark>") }
+                value.push_str(token);
+                if to_highlight { value.push_str("</mark>") }
+            } else {
+                value.push_str(token);
+            }
+        }
+    }
+}
+```
+
+#### Store
+
+```rust
+let tokenizer = Tokenizer::new(&content);
+for (pos, token_group) in tokenizer.filter_map(only_token).enumerate().take(MAX_POSITION) {
+    let position = (attr as usize * MAX_POSITION + pos) as u32;
+    for token in token_group.normalized() {
+        words_positions.entry(token.txt()).or_insert_with(SmallVec32::new).push(position);
+    }
+}
+```
+
+```rust
+// original
+for (pos, token) in simple_tokenizer(&content).filter_map(only_token).enumerate().take(MAX_POSITION) {
+    let word = token.to_lowercase();
+    let position = (attr as usize * MAX_POSITION + pos) as u32;
+    words_positions.entry(word).or_insert_with(SmallVec32::new).push(position);
+}
```

Comment:
that should be in the second section

Reply:
I don't think so because it an usage of the tokenizer and not his implementation